src/main/java/org/dependencytrack/tasks/MetricsUpdateTask.java

```diff
@@ -81,561 +96,794 @@ public void inform(final Event e) {
-            componentMetrics.setPolicyViolationsAudited(counters.policyViolationsAudited);
-            componentMetrics.setPolicyViolationsUnaudited(counters.policyViolationsUnaudited);
-            componentMetrics.setPolicyViolationsSecurityTotal(counters.policyViolationsSecurityTotal);
-            componentMetrics.setPolicyViolationsSecurityAudited(counters.policyViolationsSecurityAudited);
-            componentMetrics.setPolicyViolationsSecurityUnaudited(counters.policyViolationsSecurityUnaudited);
-            componentMetrics.setPolicyViolationsLicenseTotal(counters.policyViolationsLicenseTotal);
-            componentMetrics.setPolicyViolationsLicenseAudited(counters.policyViolationsLicenseAudited);
-            componentMetrics.setPolicyViolationsLicenseUnaudited(counters.policyViolationsLicenseUnaudited);
-            componentMetrics.setPolicyViolationsOperationalTotal(counters.policyViolationsOperationalTotal);
-            componentMetrics.setPolicyViolationsOperationalAudited(counters.policyViolationsOperationalAudited);
-            componentMetrics.setPolicyViolationsOperationalUnaudited(counters.policyViolationsOperationalUnaudited);
-            componentMetrics.setFirstOccurrence(measuredAt);
-            componentMetrics.setLastOccurrence(measuredAt);
-            LOGGER.debug("Persisting metrics for component: " + component.getUuid());
-            qm.persist(componentMetrics);
-            // Update the convenience fields in the Component object
+
             if (component.getLastInheritedRiskScore() == null ||
-                    component.getLastInheritedRiskScore() != componentMetrics.getInheritedRiskScore()) {
-                LOGGER.debug("Updating Inherited Risk Score for component: " + component.getUuid());
-                component.setLastInheritedRiskScore(componentMetrics.getInheritedRiskScore());
-                LOGGER.debug("Persisting metrics for component: " + component.getUuid());
-                qm.persist(component);
+                    component.getLastInheritedRiskScore() != counters.inheritedRiskScore) {
+                LOGGER.debug("Updating inherited risk score of component " + componentUuid);
+                trx = qm.getPersistenceManager().currentTransaction();
+                try {
+                    trx.begin();
+                    component.setLastInheritedRiskScore(counters.inheritedRiskScore);
+                    trx.commit();
+                } finally {
+                    if (trx.isActive()) {
+                        trx.rollback();
+                    }
+                }
             }
         }
-        LOGGER.debug("Completed metrics update for component: " + component.getUuid());
+
+        LOGGER.debug("Completed metrics update for component " + componentUuid + " in " +
+                DurationFormatUtils.formatDuration(new Date().getTime() - counters.measuredAt.getTime(), "mm:ss:SS"));
```

Comment:
`Date` is used here because other areas of the codebase we interact with also still use `Date`. Introducing yet another type here would just make it more confusing.

@sonatype-lift ignore